blockchain/v1/reactor.go

```diff
@@ -137,6 +141,7 @@ func (bcR *BlockchainReactor) SetLogger(l log.Logger) {
 
 // OnStart implements cmn.Service.
 func (bcR *BlockchainReactor) OnStart() error {
+	bcR.swReporter = behaviour.NewSwitcReporter(bcR.BaseReactor.Switch)
```

Comment:
How do we know the switch has been setup by the time `OnStart` is called?

Reply:
Through code inspection :) I could add a check, but probably we don't reach here with nil Switch and crash much earlier.

The switch is created in `NewNode()` by createSwitch() that adds the reactors and calls `SetSwitch()`  for each reactor  :
```
main() -> 
     NewRunNodeCmd(nodeProvider nm.NodeProvider) ->
        n, err := nodeProvider(config, logger) ->
           node.go:NewNode() -> 
              createSwitch(.., bcReactor, ...) -> 
                 sw.AddReactor("BLOCKCHAIN", bcReactor) ->
                     sw.reactors[name] = reactor
                     reactor.SetSwitch(sw)
```

Then the node is started and that causes the reactor to start:
```
main() -> 
     NewRunNodeCmd(nodeProvider nm.NodeProvider) ->
          n, err := nodeProvider(config, logger)
          err := n.Start() ->
              (n *Node) OnStart() -> 
                    n.sw.Start()
                          (sw *Switch) OnStart() -> 
                                 reactor.Start() -> 
                                     (bs *BaseService) Start() -> 
                                           bs.impl.OnStart()
```